plugins/modules/os_compatibility.py

```
@@ -0,0 +1,115 @@
+  itential.deployer.os_compatibility:
+'''
+
+RETURN = r'''
+# These are examples of possible return values, and in general should use other names for return values.
+compatible:
+    description: Is this operating system compatible with the Itential platform
+    type: bool
+    returned: always
+    sample: false
+os_version:
+    description: A normalized string representing the OS and major version
+    type: str
+    returned: always
+    sample: 'RedHat8'
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.facts.compat import ansible_facts
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict()
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # changed is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+        compatible=False,
+        os_version="",
+        os="",
+        version=""
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifications
+    if module.check_mode:
+        module.exit_json(**result)
+
+    # Get the facts from the host
+    facts = ansible_facts(module)
+
+    # Normalize the distribution and major version
+    result["os_version"] = facts["distribution"].lower() + "-" + facts["distribution_major_version"]
```

It seems a bit redundant to return the `os_version` if it's the `os` and `version` concatenated together.

---

I have removed all the return values from this except for the true|false. We can use this to fail fast when the host is not supported OS version.